L1Trigger/TrackFindingTracklet/src/TrackletCalculatorDisplaced.cc

```diff
@@ -409,43 +426,119 @@ bool TrackletCalculatorDisplaced::LLLSeeding(const Stub* innerFPGAStub,
                 zder,
                 phiderdisk,
                 rderdisk);
+  if (settings_.debugTracklet())
+    edm::LogVerbatim("Tracklet") << __LINE__ << ":" << __FILE__ << "LLL Exact values " << innerFPGAStub->isBarrel()
+                                 << middleFPGAStub->isBarrel() << outerFPGAStub->isBarrel() << " " << phi1 << ", " << z1
+                                 << ", " << r1 << ", " << phi2 << ", " << z2 << ", " << r2 << ", " << phi3 << ", " << z3
+                                 << ", " << r3 << endl;
 
   if (settings_.useapprox()) {
     phi1 = innerFPGAStub->phiapprox(phimin_, phimax_);
     z1 = innerFPGAStub->zapprox();
     r1 = innerFPGAStub->rapprox();
 
-    phi2 = outerFPGAStub->phiapprox(phimin_, phimax_);
-    z2 = outerFPGAStub->zapprox();
-    r2 = outerFPGAStub->rapprox();
+    phi2 = middleFPGAStub->phiapprox(phimin_, phimax_);
+    z2 = middleFPGAStub->zapprox();
+    r2 = middleFPGAStub->rapprox();
+
+    phi3 = outerFPGAStub->phiapprox(phimin_, phimax_);
+    z3 = outerFPGAStub->zapprox();
+    r3 = outerFPGAStub->rapprox();
   }
 
+  if (settings_.debugTracklet())
+    edm::LogVerbatim("Tracklet") << __LINE__ << ":" << __FILE__ << "LLL Approx values " << innerFPGAStub->isBarrel()
+                                 << middleFPGAStub->isBarrel() << outerFPGAStub->isBarrel() << " " << phi1 << ", " << z1
+                                 << ", " << r1 << ", " << phi2 << ", " << z2 << ", " << r2 << ", " << phi3 << ", " << z3
+                                 << ", " << r3 << endl;
+
   double rinvapprox, phi0approx, d0approx, tapprox, z0approx;
   double phiprojapprox[N_LAYER - 2], zprojapprox[N_LAYER - 2], phiderapprox[N_LAYER - 2], zderapprox[N_LAYER - 2];
   double phiprojdiskapprox[N_DISK], rprojdiskapprox[N_DISK];
   double phiderdiskapprox[N_DISK], rderdiskapprox[N_DISK];
 
   //TODO: implement the actual integer calculation
+  approxtracklet(r1,
+                 z1,
+                 phi1,
+                 r2,
+                 z2,
+                 phi2,
+                 r3,
+                 z3,
+                 phi3,
+                 take3,
+                 ndisks,
+                 rinvapprox,
+                 phi0approx,
+                 d0approx,
+                 tapprox,
+                 z0approx,
+                 phiprojapprox,
+                 zprojapprox,
+                 phiderapprox,
+                 zderapprox,
+                 phiprojdiskapprox,
+                 rprojdiskapprox,
+                 phiderdiskapprox,
+                 rderdiskapprox);
+
+  if (isfinite(phi0approx))
+    phi0approx = angle0to2pi::make0To2pi(phi0approx);
+  assert(!isfinite(phi0approx) || fabs(phi0approx) < 2 * M_PI);
```

Comment:
@gabhijith I would suggest moving all the calls to `angle0to2pi::make0To2pi` inside the `approxtracklet` and `approxproj`/`approxprojdisk` functions. I also think the `assert` is unnecessary.

Also, I know checking whether φ<sub>0</sub> is finite was necessary at certain points during debugging, but is it still necessary? If not, then we should remove it.

These comments apply to all similar instances below.